Unexpected failure when repository bump is executed and no changes are made

## Description

During the execution of https://github.com/wazuh/wazuh-qa-automation/issues/8012 it was found that the repository bumper workflow failed when no changes were necessary. This is not expected; the expected behavior should be to skip the commit changes when no commit is necessary.
```
Run git add .
On branch enhancement/wqa8012-bump-5.0.0
nothing to commit, working tree clean
Error: Process completed with exit code 1.
```
This behavior was found in the following action:
https://github.com/wazuh/wazuh-ansible/actions/runs/28590312561/job/84771921109

Maybe a previous task that will check if changes are going to be made is needed to skip the following tasks.

### Update report

The (5.x) Repository bumper workflow fails when the requested bump is already applied to the target branch (no-op bump).

Example: Run on branch 5.0.0 with inputs version=5.0.0, stage=beta3. The branch's VERSION.json was already at 5.0.0/beta3, so the bumper script modified no files and the commit step failed:

```
Run git add .
On branch enhancement/wqa8012-bump-5.0.0
nothing to commit, working tree clean
Error: Process completed with exit code 1.
```

Root cause: The Commit changes (Bump) step runs git commit unconditionally. The revert flow already guards against empty commits (git diff --staged --quiet + has_changes output); the bump flow did not.

### Update report

The repository bumper workflows (4_bumper_repository.yml / 5_bumper_repository.yml) fail when the requested bump is already applied to the target branch. The bump script runs successfully but modifies no files, and the commit step runs git commit unconditionally:

```
Run git add .
On branch enhancement/wqa8012-bump-5.0.0
nothing to commit, working tree clean
Error: Process completed with exit code 1.
```


Fix applied:

A no-changes guard was added to the commit step. When the bump produces no changes, the workflow logs it, skips commit/push/PR/merge, and finishes successfully:

```
- name: Commit changes (Bump)
  id: bump_commit
  run: |
    git add .
    if git diff --staged --quiet; then
      echo "Nothing to bump: the repository is already at the requested version/stage. Skipping commit."
      echo "has_changes=false" >> $GITHUB_OUTPUT
    else
      git commit -m "feat: bump ${{ github.ref_name }}"
      echo "has_changes=true" >> $GITHUB_OUTPUT
    fi
```


Changes per repository

|Repository	| 5.x bumper (branch 5.0.0) |	4.x bumper (branch 4.14.7)	| Branches created|
|:--|:--|:--|:--|
|wazuh-ansible	|✅ Fixed	|✅ Fixed	|bug/2185-fix-bumper-to5.0.0, bug/2185-fix-bumper-to4.14.6|

--- a/.github/workflows/5_bumper_repository.yml
+++ b/.github/workflows/5_bumper_repository.yml
@@ -143,10 +143,17 @@ jobs:
           bash ${{ env.BUMP_SCRIPT_PATH }} ${{ steps.vars.outputs.script_params }}
 
       - name: Commit changes (Bump)
+        id: bump_commit
         if: inputs.revert != true
         run: |
           git add .
-          git commit -m "feat: bump ${{ github.ref_name }}"
+          if git diff --staged --quiet; then
+            echo "Nothing to bump: the repository is already at the requested version/stage. Skipping commit."
+            echo "has_changes=false" >> $GITHUB_OUTPUT
+          else
+            git commit -m "feat: bump ${{ github.ref_name }}"
+            echo "has_changes=true" >> $GITHUB_OUTPUT
+          fi
 
       - name: Fetch full history (Revert)
         if: inputs.revert == true
@@ -197,13 +204,13 @@ jobs:
           fi
 
       - name: Push changes
-        if: inputs.revert != true || (inputs.revert == true && steps.revert_step.outputs.has_changes == 'true')
+        if: (inputs.revert != true && steps.bump_commit.outputs.has_changes == 'true') || (inputs.revert == true && steps.revert_step.outputs.has_changes == 'true')
         run: |
           git push origin ${{ steps.vars.outputs.branch_name }}
 
       - name: Create pull request
         id: create_pr
-        if: inputs.revert != true || (inputs.revert == true && steps.revert_step.outputs.has_changes == 'true')
+        if: (inputs.revert != true && steps.bump_commit.outputs.has_changes == 'true') || (inputs.revert == true && steps.revert_step.outputs.has_changes == 'true')
         run: |
           gh auth setup-git
           PR_URL=$(gh pr create \
@@ -216,7 +223,7 @@ jobs:
           echo "pull_request_url=${PR_URL}" >> $GITHUB_OUTPUT
 
       - name: Merge pull request
-        if: inputs.revert != true || (inputs.revert == true && steps.revert_step.outputs.has_changes == 'true')
+        if: (inputs.revert != true && steps.bump_commit.outputs.has_changes == 'true') || (inputs.revert == true && steps.revert_step.outputs.has_changes == 'true')
         run: |
           # Any checks for the PR are bypassed since the branch is expected to be functional (i.e. the bump process does not introduce any bugs)
           gh pr merge "${{ steps.create_pr.outputs.pull_request_url }}" --merge --admin
@@ -224,9 +231,13 @@ jobs:
       - name: Show logs
         if: inputs.revert != true
         run: |
-          echo "Bump complete."
-          echo "Branch: ${{ steps.vars.outputs.branch_name }}"
-          echo "PR: ${{ steps.create_pr.outputs.pull_request_url }}"
+          if [[ "${{ steps.bump_commit.outputs.has_changes }}" == "true" ]]; then
+            echo "Bump complete."
+            echo "Branch: ${{ steps.vars.outputs.branch_name }}"
+            echo "PR: ${{ steps.create_pr.outputs.pull_request_url }}"
+          else
+            echo "Bump skipped: the repository is already at the requested version/stage."
+          fi
           echo "Bumper scripts logs:"
           cat ${BUMP_LOG_PATH}/repository_bumper*log
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -60,6 +60,7 @@ All notable changes to this project will be documented in this file.
 
 ### Fixed
 
+- Fix bumper workflow failure when bump produces no changes ([#2195](https://github.com/wazuh/wazuh-ansible/pull/2195))
 - Bumper script issue when the tag is set to false ([#2159](https://github.com/wazuh/wazuh-ansible/issues/2159))
 - Ansible deployment fails on needrestart task when no services require restart ([#2045](https://github.com/wazuh/wazuh-ansible/issues/2045))
 - No idempotency for the `opensearch.yml` configurations ([#2056](https://github.com/wazuh/wazuh-ansible/issues/2056))

